test/Combinatorics/EnumerativeCombinatorics/partitions.jl

```diff
@@ -52,118 +55,105 @@
 
-      # Create the same by filtering all partitions
-      Q = collect(partitions(n))
-      filter!( Q->length(Q) == k, Q)
+    # Create the same by filtering all partitions
+    Q = collect(partitions(n))
+    filter!( Q->length(Q) == k, Q)
 
-      # Check that P and Q coincide (up to reordering)
-      @test length(P) == length(Q)
-      @test Set(P) == Set(Q)
+    # Check that P and Q coincide (up to reordering)
+    @test length(P) == length(Q)
+    @test Set(P) == Set(Q)
 
-      # Compare length with number_of_partitions(n,k)
-      @test length(P) == number_of_partitions(n,k)
-    end
+    # Compare length with number_of_partitions(n,k)
+    @test length(P) == number_of_partitions(n,k)
   end
 
   ############################################################################
-  # partitions(n,k,l1,l2)
+  # partitions(n,k,lb,ub)
   ############################################################################
-  for n in 0:20
-    for k = 0:n+1
-      for l1 = 0:n
-        for l2 = l1:n
-          P = collect(partitions(n,k,l1,l2))
+  @testset "partitions($n,$k,$lb,$ub)" for n in 0:20, k in 0:n+1, lb in 0:n, ub in lb:n
+    P = collect(partitions(n,k,lb,ub))
 
-          # Create the same by filtering all partitions
-          Q = collect(partitions(n,k))
-          filter!( Q->all(>=(l1),Q), Q)
-          filter!( Q->all(<=(l2),Q), Q)
+    # Create the same by filtering all partitions
+    Q = collect(partitions(n,k))
+    filter!( Q->all(>=(lb),Q), Q)
+    filter!( Q->all(<=(ub),Q), Q)
 
-          # Check that P and Q coincide (up to reordering)
-          @test length(P) == length(Q)
-          @test Set(P) == Set(Q)
-        end
-      end
-    end
+    # Check that P and Q coincide (up to reordering)
+    @test length(P) == length(Q)
+    @test Set(P) == Set(Q)
   end
 
   ############################################################################
-  # partitions(n,k,l1,l2; only_distinct_parts=true)
+  # partitions(n,k,lb,ub; only_distinct_parts=true)
   ############################################################################
-  for n in 0:20
-    for k = 0:n+1
-      for l1 = 0:n
-        for l2 = l1:n
-          P = collect(partitions(n, k, l1, l2; only_distinct_parts=true))
+  @testset "partitions($n,$k,$lb,$ub; only_distinct_parts=true)" for n in 0:20, k in 0:n+1, lb in 0:n, ub in lb:n
+    P = collect(partitions(n, k, lb, ub; only_distinct_parts=true))
 
-          # Create the same by filtering all partitions
-          Q = collect(partitions(n, k, l1, l2))
-          filter!( Q->Q==unique(Q), Q )
+    # Create the same by filtering all partitions
+    Q = collect(partitions(n, k, lb, ub))
+    filter!( Q->Q==unique(Q), Q )
 
-          # Check that P and Q coincide (up to reordering)
-          @test length(P) == length(Q)
-          @test Set(P) == Set(Q)
-        end
-      end
-    end
+    # Check that P and Q coincide (up to reordering)
+    @test length(P) == length(Q)
+    @test Set(P) == Set(Q)
```

Comment:
This loop (and probably the ones above too) creates a ton of output, see e.g.
https://github.com/oscar-system/Oscar.jl/actions/runs/9093356754/job/24992087132?pr=3389
Can this maybe be suppressed by wrapping it in another `@testset` or something similar? GitHub even refuses to give the full log initially since it is so large.

I also find this kind of a testing against itself a bit dubious if it happens at this scale.

Reply:
I had to wrap this because I was getting failing tests and it was impossible to see which ones were actually failing. I feel like I do need to see the test set names for the failing tests. But I agree with lkastner, I will split this into a nested `@testset`

Reply:
@lkastner  Okay, now this testset is split, and I also reduced the scale since I also think it's silly to test quite this many cases. Let me know if you think this is better now.